sensors/sensors.go

```diff
@@ -0,0 +1,58 @@
+// Package sensors defines interfaces for sensors used by viam cartographer
+package sensors
+
+import (
+	"context"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+	goutils "go.viam.com/utils"
+)
+
+// TimedSensorReadingResponse represents a sensor reading with a time & allows the caller to know if the reading is from a replay sensor.
+type TimedSensorReadingResponse struct {
+	Reading     []byte
+	ReadingTime time.Time
+	Replay      bool
+}
+
+// TimedSensor desribes a sensor that reports the time the reading is from & whether or not it is from a replay sensor.
+type TimedSensor interface {
+	TimedSensorReading(ctx context.Context) (TimedSensorReadingResponse, error)
+}
+
+// ValidateGetData checks every sensorValidationIntervalSec if the provided lidar
+// returned a valid timed lidar readings every sensorValidationIntervalSec
+// until either success or sensorValidationMaxTimeoutSec has elapsed.
+// returns an error if no valid lidar readings were returned.
+func ValidateGetData(
```

Comment:
same code as https://github.com/viamrobotics/viam-cartographer/pull/196/files#diff-f909251fc949bc4d5ae97f9b17ec70b8ff6ba0b2535d14aa50058ea37595a47eL84, just using the new interface